### PR TITLE
Edit feature descriptions

### PR DESCRIPTION
**Pull request by sid-kap**
Follow-up to #37. 

This is just to point out some ideas I have about how the Features section of the homepage can be improved. I know many of you have much more experience teaching Haskell to beginners, and so you would probably have more valuable input about this topic than me. 

Generally, my goal is to make these descriptions shorter and simpler. I think some of these descriptions are trying to fit way too much information in just 3-4 sentences, and the result is that it doesn't make any sense to someone who doesn't know Haskell.

### CODE REVIEW

File: src/HL/View/Home/Features.hs [hunk 1]
```diff
@@ -56,11 +54,7 @@ purefunc =
 statically :: Html ()
 statically =
   do h2_ "Statically typed"
-     p_ "Every expression in Haskell has a type which is determined at compile time. \
-       \All the types composed together by function application have to match up. If \
-       \they don't, the program will be rejected by the compiler. Types become not \
-       \only a form of guarantee, but a language for expressing the construction \
-       \of programs."
+     p_ ""
```
Comment (sid-kap) [inline src/HL/View/Home/Features.hs:57]: Most people either know what "statically-typed" means or they don't. Three sentences is not enough to explain what it is.

File: src/HL/View/Home/Features.hs [hunk 2]
```diff
@@ -115,10 +109,8 @@ concurrent =
 inference :: Html ()
 inference =
   do h2_ "Type inference"
-     p_ "You don't have to explicitly write out every type in a Haskell program. \
-       \Types will be inferred by unifying every type bidirectionally. However, you \
-       \can write out types if you choose, or ask the compiler to write them for you \
-       \for handy documentation."
+     p_ "Static types without the pain. Specify the types when you want to, let the \
+        \compiler infer the types when you don't."
```
Comment (sid-kap) [inline src/HL/View/Home/Features.hs:113]: "unifying" and "bidirectionally" are way too academic.

File: src/HL/View/Home/Features.hs [hunk 3]
```diff
@@ -153,11 +145,10 @@ inference =
...
   do h2_ "Lazy"
-     p_ "Functions don't evaluate their arguments. This means that programs \
-       \can compose together very well, with the ability to write control \
-       \constructs (such as if/else) just by writing normal functions. The purity \
-       \of Haskell code makes it easy to fuse chains of functions together, allowing \
-       \for performance benefits."
+     p_ "Values are computed when they are needed, rather than when they are defined in code. \
+        \ If a value you defined is not used, it never gets evaluated. This means you can spend \
+        \less time thinking about how your program is executed, and more time thinking about what \
+        \your program does."
```
Comment (sid-kap) [inline src/HL/View/Home/Features.hs:151]: This write-up is still too long in my opinion. Can we make it even shorter? (Idea: delete the first sentence.)

Comment (sid-kap) [inline src/HL/View/Home/Features.hs:151]: My last sentence is an attempt at explaining why laziness is a good thing. I don't think I did a great job though.

File: src/HL/View/Home/Features.hs (hunk 1 above)
Comment (chrisdone) [inline src/HL/View/Home/Features.hs:57]: Removing this text entirely will make the page look unbalanced.  I'd prefer some useful text than nothing. Please don't delete any sections without having something better to replace it. ☝️🐵

File: src/HL/View/Home/Features.hs (hunk 2 above)
Comment (chrisdone) [inline src/HL/View/Home/Features.hs:113]: Agree!

File: src/HL/View/Home/Features.hs (hunk 3 above)
Comment (chrisdone) [inline src/HL/View/Home/Features.hs:151]: I'm not sure this is an improvement. 😮  The original text explains what laziness is the way SPJ explained to a newbie crowd ("functions don't evaluate their arguments"), and covers the benefits of laziness in two more sentences ("composability, control structures" and "fusing for performance"). The suggested replacement here talks about what laziness is, and makes a vague claim about executing a program versus what the program does (I would say this won't mean much to most people, as they sound like the same thing). I'd leave this paragraph as it is until there's a better replacement. It doesn't have to be perfect.

**Comment by chrisdone** [comment]
I think our time should be spent documenting things for Haskellers, [like this](https://ci.haskell-lang.org/package/http-client) and [here](https://github.com/haskell-lang/haskell-lang/blob/master/static/markdown/package-aeson.md).

**Pull request closed without merging**